R/tidy-upkeep.R

```diff
@@ -111,6 +111,42 @@ if (year <= 2022) {
     )
   }
 
+  if (year <= 2023) {
+
+    desc <- proj_desc()
+
+    bullets <- c(bullets,
+     "2023",
+     "",
+     "Posit updates:",
+     "",
+     todo('Update copyright holder in DESCRIPTION: `person(given = "Posit, PBC", role = c("cph", "fnd"))`',
+          rstudio_pkg && !rstudio_person_ok),
+     todo("Double check license file uses '[package] authors' as copyright holder. Run `use_mit_license()`",
+          grepl("MIT", desc$get_field("License"))),
+     todo("Update email addresses *@rstudio.com -> *@posit.co",
+          any(grepl("rstudio", desc$get_authors()))),
+     todo("`usethis::use_tidy_coc()`"),
+     "",
+     "pkgdown:",
+     "",
+     todo("Update pkgdown site using instructions at <https://tidytemplate.tidyverse.org>"),
+     todo("Ensure pkgdown `development` is `mode: auto` in pkgdown config"),
+     todo("Submit PR [here](https://github.com/rstudio/aws-main/tree/main/zones) adding your site to the appropriate domain (eg., r-lib, tidyverse, tidymodels) and set url in GitHub Settings > Pages > Custom Domain"),
+     todo("Re-publish released site; see [How to update a released site](https://pkgdown.r-lib.org/dev/articles/how-to-update-released-site.html)"),
+     "",
+     todo("Modernize citation files; see updated `use_citation()`"),
+     todo("Update logo (https://github.com/rstudio/hex-stickers); run `use_tidy_logo()`"),
+     todo('Use `pak::pkg_install("org/pkg") in README'),
+     todo("Consider running `use_tidy_dependencies()` and/or replace compat files with `use_standalone()`"),
+     todo("Use `rlang::check_*` (https://github.com/r-lib/usethis/issues/1692)"),
```

Comment:
I think I need a bit more explanation as to which context this applies

Reply:
Hopefully @hadley can confirm that I have this right and whether he thinks the current spring cleaning is a good time to nudge on this.

But I think this is about placing rlang's type checking standalone file in our packages:

https://github.com/r-lib/rlang/blob/6a5a1ec1128bfeeb081d3b4815ca8e12c488a303/R/standalone-types-check.R

and calling *those* functions instead of idiosyncratic, homegrown solutions in, e.g., `R/utils.R`.

I think the payoff is getting better and more consistent errors for type errors.

Then, it looks like the remaining meat of #1692 (providing guidance on writing type checkers) might be a good issue for https://design.tidyverse.org/.

Maybe `use_tidy_dependencies()` should also place this file?

To pre-emptively answer a question: you might ask why these functions aren't in rlang itself, which most of our packages have in `Imports` already. I believe it's because we are incubating the functions in the standalone file and when they feel pretty battle-tested and stable, they might move into rlang itself.

Reply:
For this bullet, I I think you could say "`use_standalone("r-lib/rlang", "types-check")` instead of home grown argument checkers"?

Reply:
That's great context, thanks Jenny and Hadley. I'll update the bullet